### PR TITLE
[INLONG-6246][Sort] Import multiple sink option and schema update handle policy

### LINKED ISSUE #6246: [Feature][Sort] Import multiple sink option  and schema update handle policy
### Description

Import multiple sink option  and schema update handle policy

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by thesumery**
[INLONG-6246][Sort] Import multiple sink option and schema update handle policy

### Prepare a Pull Request
- Title Example: [INLONG-6246][Sort] Import multiple sink option and schema update handle policy
- Fixes #6246

### Motivation

*Import multiple sink option and schema update handle policy*

### Modifications

*Import multiple sink option and schema update handle policy*

### CODE REVIEW

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/TableChange.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+package org.apache.inlong.sort.base.sink;
+
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Objects;
+
+public interface TableChange {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/TableChange.java:28]: Add some comments?

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/SchemaUpdateExceptionPolicy.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.inlong.sort.base.sink;
+
+public enum SchemaUpdateExceptionPolicy {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/SchemaUpdateExceptionPolicy.java:21]: Add some comments?

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/MultipleSinkOption.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+
+import static org.apache.inlong.sort.base.sink.SchemaUpdateExceptionPolicy.ALERT_WITH_IGNORE;
+import static org.apache.inlong.sort.base.sink.SchemaUpdateExceptionPolicy.LOG_WITH_IGNORE;
+import static org.apache.inlong.sort.base.sink.SchemaUpdateExceptionPolicy.TRY_IT_BEST;
+
+public class MultipleSinkOption implements Serializable {
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/MultipleSinkOption.java:30]: Add some comments?

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java [hunk 4]
```diff
@@ -145,15 +146,15 @@ public final class Constants {
                     .withDescription("The option 'sink.multiple.enable' "
                             + "is used to determine whether to support multiple sink writing, default is 'false'.");
 
-    public static final ConfigOption<String> SINK_MULTIPLE_ADD_COLUMN_POLICY =
+    public static final ConfigOption<SchemaUpdateExceptionPolicy> SINK_MULTIPLE_ADD_COLUMN_POLICY =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java:149]: Add some comments?

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java [hunk 5]
```diff
@@ -145,15 +146,15 @@ public final class Constants {
...
             ConfigOptions.key("sink.multiple.add-column.policy")
-                    .stringType()
-                    .noDefaultValue()
-                    .withDescription("");
+                    .enumType(SchemaUpdateExceptionPolicy.class)
+                    .defaultValue(SchemaUpdateExceptionPolicy.TRY_IT_BEST)
+                    .withDescription("The action to deal with column add.");
 
-    public static final ConfigOption<String> SINK_MULTIPLE_DEL_COLUMN_POLICY =
+    public static final ConfigOption<SchemaUpdateExceptionPolicy> SINK_MULTIPLE_DEL_COLUMN_POLICY =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/Constants.java:155]: Add some comments?

**Review by yunqingmoswu — APPROVED** [review]

**Review by healchow — APPROVED** [review]

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/SchemaUpdateExceptionPolicy.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * Maybe one table data have different schema, once it's schema mismatch with catalog schema, how to handle
+ * this table data. For example schema mismatch:
+ *
+ * <pre>
+ * data : {a : int, b : string, c : date}
+ * catalog : {a : string, b : timestamp}
+ * </pre>
+ */
+public enum SchemaUpdateExceptionPolicy {
+    TRY_IT_BEST("Try it best to handle schema update, if can not handle it, just ignore it."),
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/SchemaUpdateExceptionPolicy.java:32]: try it best -> try best

File: inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/TableChange.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    final class DeleteColumn implements ColumnChange {
+        @Override
+        public String[] fieldNames() {
+            return new String[0];
+        }
+    }
+
+    final class UnknownColumnChange implements ColumnChange {
```
Comment (EMsnap) [inline inlong-sort/sort-connectors/base/src/main/java/org/apache/inlong/sort/base/sink/TableChange.java:181]: please add some doc

**Pull request merged**